core-java-modules/core-java-streams-3/src/main/java/com/baeldung/streams/bigdecimals/StreamAddNumbers.java

```diff
@@ -0,0 +1,35 @@
+package com.baeldung.streams.bigdecimals;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.IntStream;
+
+public class StreamAddNumbers {
+
+    public int addInts() {
+        IntStream intNumbers = new Random().ints(5, 1, 100);
+        return intNumbers.sum();
+    }
+
+    public double addDoubleOnList() {
+        List<Double> doubleNumbers = Arrays.asList(23.48, 52.26, 13.5);
+        return doubleNumbers.stream()
+            .mapToDouble(Double::doubleValue)
+            .sum();
+    }
+
+    public int addNumbersWithReduce() {
+        List<Integer> intNumbers = Arrays.asList(5, 1, 100);
```

Comment:
Use `Stream.of` rather than `asList().stream()`

Reply:
Done